sql/expression/function/sleep.go

```diff
@@ -37,8 +37,15 @@ func (s *Sleep) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
 		return nil, err
 	}
 
-	time.Sleep(time.Duration(child.(float64)*1000) * time.Millisecond)
-	return 0, nil
+	t := time.NewTimer(time.Duration(child.(float64) * 1000) * time.Millisecond)
+	defer t.Stop()
+
+	select {
+	case <-ctx.Done():
+		return context.Canceled, nil
```

Comment:
context.Canceled should be returned as the last parameter, which is the error

Reply:
Thanks, fixed.